File: example1/src/main/java/com/example/grpc/feature/waitforready/UnaryBlockingClient.java
package com.example.grpc.feature.waitforready;

import com.example.grpc.echo.EchoRequest;
import com.example.grpc.echo.EchoServiceGrpc;
import com.example.grpc.echo.Loggers;
import io.grpc.ManagedChannelBuilder;
import io.grpc.StatusRuntimeException;

import java.util.concurrent.TimeUnit;
import java.util.logging.Logger;

public class UnaryBlockingClient {

    private static final Logger logger = Logger.getLogger(UnaryBlockingClient.class.getName());

    public static void main(String[] args) throws Exception {
        Loggers.init();

        var channel = ManagedChannelBuilder.forAddress("localhost", 50051).usePlaintext().build();

        try {
            var blockingStub = EchoServiceGrpc.newBlockingStub(channel);
            var request = EchoRequest.newBuilder().setMessage("world").build();
            var response = blockingStub.unaryEcho(request);
            logger.info("response: " + response.getMessage());
        } catch (StatusRuntimeException e) {
            logger.warning("error: " + e.getStatus());
        } finally {
            channel.shutdown().awaitTermination(30, TimeUnit.SECONDS);
        }
    }
}

File: grpc-kotlin-examples/src/main/kotlin/com/example/grpc/methodtypes/unary/UnaryBlockingClient.kt
package com.example.grpc.methodtypes.unary

import com.example.grpc.EchoServiceGrpcKt
import com.example.grpc.echoRequest
import io.grpc.ManagedChannel
import io.grpc.ManagedChannelBuilder
import io.grpc.examples.helloworld.GreeterGrpcKt.GreeterCoroutineStub
import io.grpc.examples.helloworld.helloRequest
import java.io.Closeable
import java.util.concurrent.TimeUnit

class UnaryBlockingClient(private val channel: ManagedChannel) : Closeable {
  private val stub: EchoServiceGrpcKt.EchoServiceCoroutineStub = EchoServiceGrpcKt.EchoServiceCoroutineStub(channel)

  suspend fun greet(name: String) {
    val request = echoRequest { this.message = name }
    val response = stub.unaryEcho(request)
    println("Received: ${response.message}")
  }

  override fun close() {
    channel.shutdown().awaitTermination(5, TimeUnit.SECONDS)
  }
}

/** Greeter, uses first argument as name to greet if present; greets "world" otherwise. */
suspend fun main(args: Array<String>) {
  val port = 50051

  val channel = ManagedChannelBuilder.forAddress("localhost", port).usePlaintext().build()

  val client = UnaryBlockingClient(channel)

  val user = "world"
  client.greet(user)
}
